### PR TITLE
[DIT-4693] (v4) Improve error handling 

### PR DESCRIPTION
**Pull request by azjgard**
## Overview
- migrate to esbuild
- integrate Sentry
- smoothly handle errors to prevent giant stack traces from flooding console

## Context
https://linear.app/dittowords/issue/DIT-4693/improve-cli-error-handling-and-reporting

## Test Plan
- [x] Install deps `yarn install`
- [x] Start your local API server in `ditto-app`: `yarn start:api`
- [x] Point your CLI at the local API server: `export DITTO_API_HOST=http://localhost:3001 ` (in your CLI directory)
### The CLI still functions with esbuild
1. Run `yarn start`
2. Confirm that output files are written to `ditto/`
### The CLI handles errors
1. Hardcore an error somewhere in `pull.ts`
2. Run `yarn start`
3. Confirm that you get a suave error message without a stack trace
4. Confirm that your hardcoded error shows up in Sentry

## Jordin to test
- [x] Published `v4.0.0-alpha.0` to confirm that CLI still runs with new esbuild setup
- [x] Published `v4.0.0-alpha.1` to confirm that Sentry errors are appropriately captured

### CODE REVIEW

File: lib/init/project.ts [hunk 1]
```diff
@@ -105,7 +105,7 @@ export const collectAndSaveSource = async (
     const token = config.getToken(consts.CONFIG_FILE, consts.API_HOST);
     const project = await collectSource(token, components);
     if (!project) {
-      quit("", 0);
```
Comment (azjgard) [inline lib/init/project.ts:108]: These changes are just for clearer intent, no functional difference

File: lib/utils/createSentryContext.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+type SentryContext = Record<
+  string,
+  string | number | boolean | null | undefined
+>;
+
+/**
+ * Sentry context
```
Comment (azjgard) [inline lib/utils/createSentryContext.ts:7]:
```suggestion
 * Sentry context only supports properties one layer deep
```

File: lib/utils/quit.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
-export function quit(message: string, exitCode = 2) {
-  console.log(`\n${message}\n`);
+export function quit(message: string | null, exitCode = 2) {
+  if (message) console.log(`\n${message}\n`);
   process.exitCode = exitCode;
-  process.exit();
```
Comment (azjgard) [inline lib/utils/quit.ts:4]: Explicitly exiting the process like this is bad, and can kill network requests in-flight (like captured exceptions being sent to Sentry)

File: package.json [hunk 4]
```diff
@@ -1,15 +1,15 @@
 {
   "name": "@dittowords/cli",
-  "version": "3.10.1",
+  "version": "4.0.0-alpha.0",
   "description": "Command Line Interface for Ditto (dittowords.com).",
   "license": "MIT",
   "main": "bin/index.js",
   "scripts": {
-    "prepublish": "tsc",
+    "prepublish": "etsc && sentry-cli sourcemaps inject ./bin && npx sentry-cli sourcemaps upload ./bin --release=\"$(cat package.json | jq -r '.version')\"",
```
Comment (kadenbarlow) [inline package.json:8]: updating this to prepublish:only

**Review by kadenbarlow — APPROVED** [review]
We found one bug where yarn start doesn't automatically run pull in dev environment and will create another ticket for it. 
Everything else worked great.

Recommended adding in the sentry error id into the message for easier support responses.

**Pull request merged**